mods/dungeon_loot/mapgen.lua

```diff
@@ -0,0 +1,159 @@
+			local node = get_node(pos)
+			if is_wall(node) then
+				local front_below = vector.subtract(pos, {x=dir.x, y=1, z=dir.z})
+				local above = vector.add(pos, {x=0, y=1, z=0})
+
+				-- check that it:
+				--- is at least 2 nodes high (not a staircase)
+				--- has a floor
+				if is_wall(get_node(front_below)) and is_wall(get_node(above)) then
+					table.insert(ret, {pos=pos, facing={x=-dir.x, y=0, z=-dir.z}})
+					if dir.z == 0 then
+						mindist.x = min(mindist.x, i-1)
+					else
+						mindist.z = min(mindist.z, i-1)
+					end
+					wallnode = node.name
+				end
+				-- abort even if it wasn't a wall cause something is in the way
+				break
+			end
+		end
+	end
+
+	return {
+		walls = ret,
+		size = {x=mindist.x*2, z=mindist.z*2},
+		type = wallnode == wall_sand and "desert" or "normal"
+	}
+end
+
+local function populate_chest(pos, rand, dungeontype)
+	-------------------- COMMENT THESE OUT BEFORE MERGING --------------------
+	minetest.chat_send_all("chest placed at " .. minetest.pos_to_string(pos) .. " [" .. dungeontype .. "]")
+	minetest.add_node(vector.add(pos, {x=0, y=1, z=0}), {name="default:torch", param2=1})
+	-------------------- COMMENT THESE OUT BEFORE MERGING --------------------
+
+	local item_list = dungeon_loot._internal_get_loot(pos.y, dungeontype)
+	-- take random (partial) sample of all possible items
+	assert(#item_list >= dungeon_loot.STACKS_PER_CHEST_MAX)
+	item_list = random_sample(rand, item_list, dungeon_loot.STACKS_PER_CHEST_MAX)
+
+	-- apply chances / randomized amounts and collect resulting items
+	local items = {}
+	for _, loot in ipairs(item_list) do
+		if rand:next(0, 1000) / 1000 <= loot.chance then
+			local itemdef = minetest.registered_items[loot.name]
+			local amount = 1
+			if loot.count ~= nil then
+				amount = rand:next(loot.count[1], loot.count[2])
+			end
+
+			if itemdef.tool_capabilities then
+				for n = 1, amount do
+					local wear = rand:next(0.20 * 65535, 0.75 * 65535) -- 20% to 75% wear
+					table.insert(items, ItemStack({name=loot.name, wear=wear}))
+				end
+			elseif itemdef.stack_max == 1 then
+				-- not stackable, add separately
+				for n = 1, amount do
+					table.insert(items, loot.name)
+				end
+			else
+				table.insert(items, ItemStack({name=loot.name, count=amount}))
+			end
+		end
+	end
+
+	-- place items at random places in chest
+	local inv = minetest.get_meta(pos):get_inventory()
+	local listsz = inv:get_size("main")
+	assert(listsz >= #items)
+	for _, item in ipairs(items) do
+		local index = rand:next(1, listsz)
+		if inv:get_stack("main", index):is_empty() then
+			inv:set_stack("main", index, item)
+		else
+			inv:add_item("main", item) -- space occupied, just put it anywhere
+		end
+	end
+end
+
+
+minetest.register_on_generated(function(minp, maxp, blockseed)
+	local poslist = minetest.get_mapgen_object("gennotify")["dungeon"]
+	if poslist == nil then return end
+
+	local noise = minetest.get_perlin(10115, 4, 0.5, 1)
```

Comment:
Please use 1 octave for noise, the visual quality of the noise is unimportant for chests.